### PR TITLE
INTER-47: Designed new UI Modal component

### PR DESCRIPTION
**Pull request by paul4702**
image
image
image

### CODE REVIEW

File: src/lib/atoms/v2/buttons/Button.dom.spec.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import Button from './Button.svelte';
+import { describe, it, afterEach } from 'vitest';
+import { cleanup, render, screen, fireEvent } from '@testing-library/svelte';
+import TestButton from './TestButton.svelte';
+
+describe('Button', () => {
```
Comment (nayanmarlin) [inline src/lib/atoms/v2/buttons/Button.dom.spec.ts:6]: @souvikmishra - Can we Remove this duplicate spec(test) files from our v2? Ultimately, The snapshot or test cases will break in some case as we are developing new UI.

File: src/lib/atoms/v2/buttons/Button.svelte [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	export let size: ButtonModel['size'] = 'medium';
+	export let styleClass = '';
+	export let onclick: ButtonModel['onclick'] = undefined;
+	export let disabled = false;
+	export let loading = false;
+
+	const buttonStyles = getButtonStyles(variant);
+	const buttonSize = getButtonSize(size);
+
+	$: buttonStyleClass = `${styleClass} ${buttonSize} ${buttonStyles} gap-1.5 !rounded-[100.86px]`;
```
Comment (nayanmarlin) [inline src/lib/atoms/v2/buttons/Button.svelte:15]: @paul4702 - Can we use `cn` function here for such CSS?

File: src/lib/atoms/v2/buttons/InfoButtonLink.svelte [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+	export let href = '';
+	export let text: string;
+	export let styleClass = '';
+
+	$: buttonClass = `${buttonClasses.lightblueFilled} ${styleClass}`;
+</script>
+
+<a
+	class="text-small mx-auto mt-4 h-10 w-fit gap-3 font-medium {buttonClass}"
```
Comment (nayanmarlin) [inline src/lib/atoms/v2/buttons/InfoButtonLink.svelte:13]: @paul4702 - same here `cn`

**Review by nayanmarlin — CHANGES_REQUESTED** [review]

File: src/lib/atoms/v2/buttons/Button.svelte (hunk 2 above)
Comment (vishwajeetraj11) [reply src/lib/atoms/v2/buttons/Button.svelte:15]: or this syntax

"{styleClass} {buttonSize} {buttonStyles} gap-1.5 !rounded-[100.86px]"

**Review by vishwajeetraj11 — APPROVED** [review]

**Review by nayanmarlin — APPROVED** [review]

**Pull request merged**